[TRRFFDA-480] Minimise the number of default registration fields

ScriptSmith opened the pull request.

ScriptSmith committed:
Remove location details from default registration forms

alintreznai commented:
@ScriptSmith I rebased the branch based on latest changes in next_release.

alintreznai commented:
Looks good overall. I'd also remove relevant fields from the html templates, for example I'd remove the `templates/registration/registration_patient_address.html` template and remove fields from `templates/registration/registration_form_patient_with_parent.html`

ScriptSmith commented:
Good point, I didn't know about those templates. Thanks!

ScriptSmith committed:
Remove address fields and logic from Parent+Patient registration

sztamas committed:
Registration field minimalisation changes:

TRRFFDA-480

  * phone number not required on registration
  * removed more unused code
  * removed hard-coded Clinician form from Parent view
  * fixed save method in ParentGuardian
  * removed code passing in context_ids to Parent views as contexts are
    per Patient not Parent (that can have multiple Patient with different
    contexts)
